storage/tick.go

```diff
@@ -85,9 +93,19 @@ func newTickManager(database database, opts Options) databaseTickManager {
 		c:        context.NewCancellable(),
 		tokenCh:  tokenCh,
 	}
+
+	runtimeOptsMgr := opts.RuntimeOptionsManager()
+	runtimeOptsMgr.RegisterListener(mgr)
+	return mgr
+}
+
+func (mgr *tickManager) SetRuntimeOptions(opts runtime.Options) {
```

Comment:
Why is this updateable after instantiation? Is there a KV config that I missed

Reply:
also seems a little weird that this function accepts a runtime.Options and uses hat to update a single field instead of just wholesale replacing the internal object reference (which seems to be of a different type?). Could we just use the same type (and wrap it in a mutex if we need the lock)